articles/quickstart/native/ionic-angular/01-login.md

```diff
@@ -0,0 +1,245 @@
+- `clientId`: The "client ID" value present under the "Settings" of the application you created in your Auth0 dashboard
+- `redirectUri`: The URL to where you'd like to redirect your users after they authenticate with Auth0.
+- `useRefreshTokens`: The Angular SDK will use refresh tokens over relying on third-party cookies, which can be blocked in many modern browsers. Please read [Refresh Token Rotation](https://auth0.com/docs/tokens/refresh-tokens/refresh-token-rotation) for more information on using refresh tokens with public clients.
+- `cacheLocation`: The location at which to store tokens. We use `localstorage` here so that tokens are persisted across app refreshes.
+
+<%= include('../\_includes/ionic/\_note_storage') %>
+
+:::panel Checkpoint
+Now that you have configured your app with the Auth0 Angular SDK, run your application to verify that the SDK is initializing without error and that your application runs as it did before.
+:::
+
+## Add Login to Your Application
+
+<%= include('../\_includes/ionic/\_add_login_intro') %>
+
+Add a new `LoginButton` component to your application with the following code:
+
+```js
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from '@auth0/auth0-angular';
+import { Browser } from '@capacitor/browser';
+import { mergeMap } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-login-button',
+  template: `<ion-button (click)="login()">Login</ion-button>`,
+})
+export class LoginButtonComponent {
+  constructor(public auth: AuthService) {}
+
+  login() {
+    this.auth
+      .buildAuthorizeUrl()
+      .pipe(mergeMap((url) => Browser.open({ url, windowName: '_self' })))
+      .subscribe();
+  }
+}
+```
+
+This component:
+
+- defines a template with a simple button that logs the user in when clicked
+- uses `buildAuthorizeUrl` to construct a URL to Auth0's Universal Login page
+- uses Capacitor's Browser plugin to open the URL and show the login page to the user
+
+### Handling the callback
+
+<%= include('../\_includes/ionic/\_handle_callback_intro') %>
+
+Modify your `App` component and use the `ngOnInit` method to handle the callback from Auth0:
+
+```js
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from '@auth0/auth0-angular';
+import { mergeMap } from 'rxjs/operators';
+import { Browser } from '@capacitor/browser';
+import { App } from '@capacitor/app';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: 'app.component.html',
+  styleUrls: ['app.component.scss'],
+})
+export class AppComponent implements OnInit {
+  // Import the AuthService module from the Auth0 Angular SDK
+  constructor(public auth: AuthService) {}
+
+  ngOnInit(): void {
+    // Use Capacitor's App plugin to subscribe to the `appUrlOpen` event
+    App.addListener('appUrlOpen', ({ url }) => {
```

Comment:
@stevehobbsdev when does this even fire? I didn't see it running during login/logout in the Angular app.

Reply:
This fires when Auth0 redirects back to the app after login using `redirect_uri`. The redirect happens in the system browser, then Android/iOS brings the app back to the foreground thanks to the custom URL scheme being used.

This event is fired by Capacitor's App plugin (shown for Android here: https://github.com/ionic-team/capacitor-plugins/blob/main/app/android/src/main/java/com/capacitorjs/plugins/app/AppPlugin.java#L132). iOS will have a similar piece of code.